### PR TITLE
Allow multiple routing keys for sources.

### LINKED ISSUE #24: Multiple routing keys
Hey guys! It would be really handy if we could specify multiple routing keys 👍

### PR DESCRIPTION
**Pull request by adamlc**
This update will allow multiple routing keys for sources. Existing routingKey config item gets merged into the new routingKeys property, so existing support is maintained :)

Fixes #24

### CODE REVIEW

**Comment by adamlc** [comment]
Just noticed I'm getting a panic at runtime, quite new to go so I'm going to try and figure it out :)

**Comment by adamlc** [comment]
Ok so with my first commit I was getting a runtime panic:

```
{"level":"info","mappingFile":"/Users/mooloop/go/src/github.com/AirHelp/rabbit-amazon-forwarder/examples/rabbit_to_lambda.json","msg":"Loading mapping file","time":"2018-04-26T14:14:06+01:00"}
{"level":"info","msg":"Loading consumer - forwarder pairs","time":"2018-04-26T14:14:06+01:00"}
{"consumerName":"test-rabbit","consumerType":"RabbitMQ","level":"info","msg":"Creating consumer","time":"2018-04-26T14:14:06+01:00"}
{"forwarderName":"test-lambda","forwarderType":"Lambda","level":"info","msg":"Creating forwarder","time":"2018-04-26T14:14:06+01:00"}
{"forwarderName":"test-lambda","level":"info","msg":"Created forwarder","time":"2018-04-26T14:14:06+01:00"}
panic: runtime error: hash of unhashable type rabbitmq.Consumer

goroutine 1 [running]:
github.com/AirHelp/rabbit-amazon-forwarder/mapping.Client.Load(0x14aa3a0, 0x16c5570, 0x0, 0x14aa3a0, 0x16c5570)
	/Users/mooloop/go/src/github.com/AirHelp/rabbit-amazon-forwarder/mapping/mapping.go:63 +0x324
main.main()
	/Users/mooloop/go/src/github.com/AirHelp/rabbit-amazon-forwarder/server.go:18 +0x61
```

It seems that changing the routingKeys to a string slice broke the structure used for consumer and forwader mapping(`map[consumer.Client]forwarder.Client`). So to fix the issue and make things a little simpler I've created a mapping.ConsumerForwarderMap struct, as you can see in the commits.

Hope that makes sense!

File: supervisor/supervisor.go [hunk 1]
```diff
@@ -34,25 +34,25 @@ type consumerChannel struct {
...
 }
 
 // Start starts supervisor
 func (c *Client) Start() error {
 	c.consumers = make(map[string]*consumerChannel)
-	for consumer, forwarder := range c.mappings {
-		channel := makeConsumerChannel(forwarder.Name())
-		c.consumers[forwarder.Name()] = channel
-		go consumer.Start(forwarder, channel.check, channel.stop)
+	for _, forwarderMap := range c.mappings {
```
Comment (filiphaftek) [inline supervisor/supervisor.go]: Something like `mappingEntry` would be better.

File: mapping/mapping.go [hunk 2]
```diff
@@ -33,6 +34,12 @@ type Helper interface {
 	createForwarder(entry config.AmazonEntry) forwarder.Client
 }
 
+// ConsumerForwarderMap mapping for consumers and forwarders
+type ConsumerForwarderMap struct {
```
[the remark this answers is not in the document]
Comment (adamlc) [reply mapping/mapping.go]: @filiphaftek any suggestions? Sorry its been awhile since I worked on this!

[the remark this answers is not in the document]
Comment (filiphaftek) [reply mapping/mapping.go]: ConsumerForwarderMapping would be better imho, as it's self describing.

**Comment by adamlc** [comment]
@filiphaftek I've renamed the items you requested 👍

**Comment by filiphaftek** [comment]
hi @adamlc , sorry that I took me so long to verify this. The problem is that when I am tetsing e2e with previous approach with `routing`, the binding in RabbitMQ is created 
image but when using new approach `"routingKeys" : ["test3", "test4"]` I have binding not fulfilled: 
image. Could you test it on really RabbitMQ with multiple routing key? Or maybe am I missing something? Thank you.

**Comment by filiphaftek** [comment]
@adamlc any update on this?

**Comment by adamlc** [comment]
@filiphaftek I can't replicate this at all, with the following config:

```json
[
  {
    "source" : {
      "type" : "RabbitMQ",
      "name" : "test-rabbit",
      "connection" : "amqp://guest:guest@localhost:5672/",
      "topic" : "amq.topic",
      "queue" : "test-queue",
      "routingKeys" : ["#", "test3", "test4"]
    },
    "destination" : {
      "type" : "SNS",
      "name" : "test-sns",
      "target" : "arn:aws:sns:eu-west-1:XXXXXXXX:test-forwarder"
    }
  }
]
```

The bindings seem to work fine for me building from my branch.

rabbitmq management

Are you building from the multiple-routing-keys branch?

**Comment by filiphaftek** [comment]
@adamlc my bad, tested e2e today both scenarios and all is good. Please merge our master branch  to your fork and we are good to merge it. Thank you very much for great contribution 👍

**Comment by adamlc** [comment]
@filiphaftek no problem! I've rebased with master 👍

**Pull request merged**